app/controllers/admin/api/services/proxies_controller.rb

```diff
@@ -58,7 +58,9 @@ def show
   #
   def update
     if proxy.update_attributes(proxy_params)
-      proxy.deploy_v2 if proxy.apicast_configuration_driven
+      if proxy.apicast_configuration_driven
```

Comment:
Imo it is easier to read with less indented ifs, like:
```ruby
if proxy.update_attributes(proxy_params) && proxy.apicast_configuration_driven
  proxy.service_mesh_integration? ? proxy.deploy! : proxy.deploy_v2
end
```

or even:
```ruby
return if !proxy.update_attributes(proxy_params) || !proxy.apicast_configuration_driven
proxy.service_mesh_integration? ? proxy.deploy! : proxy.deploy_v2
```

Reply:
Or

```ruby
case
when proxy.apicast_configuration_driven
  proxy.deploy_v2
when proxy.service_mesh_inegration
 proxy.deploy!
end

```

Reply:
I was also thinking if we could just use `proxy.deploy!` and done